header-rewrite-filter/header_processor.cc

```diff
@@ -158,6 +205,74 @@ namespace HeaderRewriteFilter {
+        // TODO: validate start pointer
+        const Utility::BooleanOperatorType start_type = Utility::StringToBooleanOperatorType(*start);
+
+        // conditional can't start with a binary operator
+        if (Utility::isBinaryOperator(start_type)) {
+            return absl::InvalidArgumentError("invalid condition -- condition must begin with '!' or an operand");
+        }
+
+        for (auto it = start; it != operation_expression.end();) {
+            const Utility::BooleanOperatorType operator_type = Utility::StringToBooleanOperatorType(*it);
+
+            // condition can't have two binary operators in a row
+            if (it != start) {
+                const Utility::BooleanOperatorType prev_operator_type = Utility::StringToBooleanOperatorType(*(it-1));
+                if (Utility::isBinaryOperator(operator_type) && Utility::isBinaryOperator(prev_operator_type)) {
+                    return absl::InvalidArgumentError("invalid condition -- cannot have two binary operators in a row");
+                }
+            }
+
+            // condition can't end with an operator
+            if (it+1 == operation_expression.end() && Utility::isOperator(operator_type)) {
+                return absl::InvalidArgumentError("invalid condition -- cannot have an operator at the end of the condition");
+            }
+
+            // parse operation type
+            if (Utility::isBinaryOperator(operator_type)) {
+                operators_.push_back(operator_type);
+                it++;
+            } else if (operator_type == Utility::BooleanOperatorType::Not) {
+                if (it + 1 == operation_expression.end()) {
+                    return absl::InvalidArgumentError("invalid condition -- must have operand after 'not'");
+                } else if (Utility::isOperator(Utility::StringToBooleanOperatorType(*(it+1)))) {
+                    return absl::InvalidArgumentError("invalid condition -- can't have an operator after 'not'");
+                }
+                operands_.push_back(std::pair<absl::string_view, bool>(*(it+1), true));
+                it += 2;
+            } else {
+                operands_.push_back(std::pair<absl::string_view, bool>(*it, false));
+                it++;
+            }
+        }
+
+        // TODO: validate number of operands and operators
+        return absl::OkStatus();
+    }
+
+    bool ConditionProcessor::executeOperation() {
+        if (operands_.size() == 1) {
+            return operands_.at(0).second ? false : true; // TODO: remove mock value
+        }
+
+        bool result = true;
+
+        // evaluate first operation
+        result = Utility::evaluateExpression(operands_.at(0), operators_.at(0), operands_.at(1));
+
+        auto operators_it = operators_.begin() + 1;
+        auto operands_it = operands_.begin() + 2;
+
+        while (operators_it != operators_.end() && operands_it != operands_.end()) {
```

Comment:
It might be useful to have an explicit priority e.g. AND is evaluated before ORs instead of evaluating based on the given input order

Reply:
I prefer to disallow ambiguous operators. If we add parenthesis, which would be super-useful anyway, then we can mandate their use for any otherwise ambiguous conditionals, e.g. `foo and bar or baz` would be rejected as invalid, operators must either `foo and (bar or baz)` or `(foo and bar) or baz`.

That makes it immediately obvious to any readers, people reading and writing the DSL don't have to learn the precedence order, look up docs, etc

Reply:
Will do, I'll work on this as a separate PR.